src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.cs

```diff
@@ -791,6 +764,216 @@ public IEnumerable<TResult> OfType<TResult>()
+        public void CopyTo(Span<T> destination)
+        {
+            var self = this;
+            self.ThrowNullRefIfNotInitialized();
+            Requires.Range(self.Length <= destination.Length, nameof(destination));
+
+            self.AsSpan().CopyTo(destination);
+        }
+
+        /// <summary>
+        /// Inserts the specified values at the specified index.
+        /// </summary>
+        /// <param name="index">The index at which to insert the value.</param>
+        /// <param name="items">The elements to insert.</param>
+        /// <returns>The new immutable collection.</returns>
+        public ImmutableArray<T> InsertRange(int index, T[] items)
+        {
+            var self = this;
+            self.ThrowNullRefIfNotInitialized();
+            Requires.Range(index >= 0 && index <= self.Length, nameof(index));
+            Requires.NotNull(items, nameof(items));
+
+            if (items.Length == 0)
+            {
+                return self;
+            }
+            if (self.IsEmpty)
+            {
+                return new ImmutableArray<T>(items);
+            }
+
+            return self.InsertSpanRangeInternal(index, items);
+        }
+
+        /// <summary>
+        /// Inserts the specified values at the specified index.
+        /// </summary>
+        /// <param name="index">The index at which to insert the value.</param>
+        /// <param name="items">The elements to insert.</param>
+        /// <returns>The new immutable collection.</returns>
+        public ImmutableArray<T> InsertRange(int index, ReadOnlySpan<T> items)
+        {
+            var self = this;
+            self.ThrowNullRefIfNotInitialized();
+            Requires.Range(index >= 0 && index <= self.Length, nameof(index));
+
+            if (items.IsEmpty)
+            {
+                return self;
+            }
+            if (self.IsEmpty)
+            {
+                return items.ToImmutableArray();
+            }
+
+            return self.InsertSpanRangeInternal(index, items);
+        }
+
+        /// <summary>
+        /// Removes the specified values from this list.
+        /// </summary>
+        /// <param name="items">The items to remove if matches are found in this list.</param>
+        /// <param name="equalityComparer">
+        /// The equality comparer to use in the search.
+        /// </param>
+        /// <returns>
+        /// A new list with the elements removed.
+        /// </returns>
+        public ImmutableArray<T> RemoveRange(ReadOnlySpan<T> items, IEqualityComparer<T>? equalityComparer = null)
+        {
+            var self = this;
+            self.ThrowNullRefIfNotInitialized();
+
+            if (items.IsEmpty || self.IsEmpty)
+            {
+                return self;
+            }
+
+            if (items.Length == 1)
+            {
+                return self.Remove(items[0], equalityComparer);
+            }
+
+#nullable disable
```

Comment:
Why do you need to disable nullable annotations for this line? I thought the `equalityComparer` argument could be nullable.

Reply:
Because `TKey `of Dictionary<,> has `not-null` constraint, but `T` of ImmutableArray has no this constraint. Disable nullable annotations here to pass compilation validation.

Reply:
Is it possible to suppress the warning using just a `!`?